src/main/kotlin/skytils/skytilsmod/utils/apiTools.kt

```diff
@@ -141,4 +141,22 @@ object BlockPosObjectSerializer : KSerializer<BlockPos> {
             }
             BlockPos(x, y, z)
         }
+}
+
+/**
+ * Serializes/deserializes a [SkyblockIsland] based on its [SkyblockIsland.mode] to retain compatibility
+ * with older saves that used the mode instead of a stringified enum constant name.
+ */
+object SkyblockIslandModeSerializer : KSerializer<SkyblockIsland> {
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("SkyblockIsland", PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): SkyblockIsland = decoder.decodeString().let { str ->
+        val sbIsland = SkyblockIsland.values().find { it.mode == str }
+        // Migrate old saves from the Blazing Fortress to the Crimson Isle
```

Comment:
won't the coordinates be outdated?

Reply:
What's the best way to update old waypoints? Right now, if there isn't any special logic, importing waypoints from the blazing fortress will throw an error because it's not present in the `SkyblockIsland` enum anymore... should I make the serializer return a nullable type? Should blazing fortress waypoints just be thrown away?

Reply:
Is it worth decoding each waypoint as a jsonelement first and then decode the jsonelement to a Waypoint while catching errors? That will prevent the whole system from failing if one imported/saved waypoint doesn't match.

I don't think it's worth keeping old waypoints though because the update changed the entire landscape